### ISSUE TITLE
False Positive: Implementations that call `super.clone()` indirectly through a helper are still treated as missing the superclass clone call.

### ISSUE THREAD

**Issue by Carlson-JLQ** [checkstyle/checkstyle#19391]
# False Positive: Implementations that call `super.clone()` indirectly through a helper are still treated as missing the superclass clone call.

Version
unknown in repository metadata

## Checker
- Checker id: `SuperCloneCheck`
- Checker description: Detects overriding clone() methods that do not contain a call to super.clone().

## Description of the false positive
`diff/output` marks 1 false positives for `SuperCloneCheck`. Across these samples, the checker tends to fire on the outer syntax while missing code-level cues that make the pattern harmless, exempted, or at least much weaker than a real bug. The recurring themes are suppression or annotations.

### `NegCase1.java`
Source: `output/codeql/MissingCallToSuperClone/src/main/java/scensct/core/neg/NegCase1.java`

The sample keeps the benchmark's target pattern but expresses it in a slightly non-canonical source form. A human reader would likely classify this as an allowed or intentionally structured pattern rather than a bug. That matches the benchmark's stated intent: A subclass's clone method does not call super.clone() on its parent class's clone method should not be flagged as the parent's clone method itself does not call super.clone().

```java
// A subclass's clone method does not call super.clone() on its parent class's clone method should not be flagged as the parent's clone method itself does not call super.clone().
package scensct.core.neg;

public class NegCase1 {
    static class Parent {
        // Parent clone method does not call super.clone() (e.g., Object.clone() is not called)
        @Override
        protected Object clone() throws CloneNotSupportedException {
            return new Parent(); // Direct instantiation, not using super.clone()
        }
    }
    
    static class Subclass extends Parent {
        @Override
        protected Object clone() throws CloneNotSupportedException {
            // This subclass clone method does not call super.clone()
            return new Subclass(); // Direct instantiation, scenario logic satisfied
        }
    }
    
    public static void main(String[] args) throws CloneNotSupportedException {
        Subclass obj = new Subclass();
        Subclass cloned = (Subclass) obj.clone();
    }
}
```

## Cause analysis
The rule is documented as follows: Detects overriding clone() methods that do not contain a call to super.clone().

In this set the problem shows up around suppression or annotations. The implementation appears to lean too heavily on surface indicators and not enough on what the code actually does.

That is why these reports read like over-approximation. The code often retains a suspicious signature or control-flow skeleton, but once you read the body, the benchmark's claimed exemption is visible in the source itself.

For `SuperCloneCheck`, the practical improvement would be to model those source-level exemptions more explicitly instead of reporting every syntactic near-match.

## References
None known.

**Comment by romani** [comment]
please follow issue template strictly

**Comment by Srvankit** [comment]
Hi, I’d like to work on this issue. Please let me know if there are any specific expectations before I begin.
